tumblr/TMTumblrSDK
# Adds support for Swift Package Manager

## What it does

- Adds a `Package.swift` file with 3 targets:

-- `TMTumblrSDK` 
-- `TMTumblrSDKTests`
-- `TMTumblrSDKSwiftTests`

- Adds missing imports of `Foundation` and `UIKit`
- Moves header files into an `include/` folder to be compatible with SPM.
- Moves the resource files into a `Resources/` folder to be compatible with SPM.
- Adds a category for `NSBundle` to help us get the correct bundle when running tests with/without SPM.
- Adds a swift rewrite of `TMBaseTestCase` to be used by `TMTumblrSDKSwiftTests`.

## How to test

- Open the `Package.swift` file to open the swift package in Xcode.
- Build and run tests. All the tests should pass. (I am getting this 2 errors locally but Im not sure if its my local setup. I get them in the master branch too):
<img width="422" alt="Screenshot 2024-04-01 at 11 09 26 a m" src="https://github.com/tumblr/TMTumblrSDK/assets/5721324/4c4aac5e-21e7-47b7-8f43-87db739b0c5b">

- Go to `TMTumblrSDK/ExampleiOS` and run `pod install`.
- Open the Xcode workspace.
- Build and run all tests. They should pass. (except for the 2 mentioned above if you also see those 2 failures).

**Note:** I didn't touch `ExampleMacOS` because it says "SWIFT_VERSION '3.0' is unsupported, supported versions are: 4.0, 4.2, 5.0."
